tor/core/user_interaction.py

```diff
@@ -226,79 +205,70 @@ def process_done(post: Comment, cfg: Config, override=False, alt_text_trigger=Fa
 
 
 def process_unclaim(post: Comment, cfg: Config) -> None:
-    # Sometimes people need to unclaim things. Usually this happens because of
-    # an issue with the post itself, like it's been locked or deleted. Either
-    # way, we should probably be able to handle it.
-
-    # Process:
-    # If the post has been reported, then remove it. No checks, just do it.
-    # If the post has not been reported, attempt to load the linked post.
-    #   If the linked post is still up, then reset the flair on ToR's side
-    #    and reply to the user.
-    #   If the linked post has been taken down or deleted, then remove the post
-    #    on ToR's side and reply to the user.
-
-    top_parent = post.submission
-
-    unclaim_failure_post_already_completed = i18n['responses']['unclaim']['post_already_completed']
-    unclaim_still_unclaimed = i18n['responses']['unclaim']['still_unclaimed']
-    unclaim_success = i18n['responses']['unclaim']['success']
-    unclaim_success_with_report = i18n['responses']['unclaim']['success_with_report']
-    unclaim_success_without_report = i18n['responses']['unclaim']['success_without_report']
+    """
+    Process an unclaim request.
 
-    # WAIT! Do we actually own this post?
-    if top_parent.author.name not in __BOT_NAMES__:
-        log.info('Received `unclaim` on post we do not own. Ignoring.')
+    Note that this function also checks whether a post should be removed and
+    does so when required.
+    """
+    submission = post.submission
+    if submission.author.name not in __BOT_NAMES__:
+        log.debug("Received 'unclaim' on post we do not own. Ignoring.")
         return
 
-    if flair.unclaimed in top_parent.link_flair_text:
-        post.reply(_(unclaim_still_unclaimed))
-        return
+    response = cfg.blossom.get_submission(reddit_id=submission.fullname)
+    if response.status != BlossomStatus.ok:
+        # If we are here, this means that the current submission is not yet in Blossom.
+        # TODO: Create the Submission in Blossom and try this method again.
+        raise Exception(f"The Submission {submission.fullname} is not present in Blossom.")
 
-    for item in top_parent.user_reports:
-        if not item[0]:
-            continue
-        if (
-            reports.original_post_deleted_or_locked in item[0]
-                or reports.post_violates_rules in item[0]
-        ):
-            top_parent.mod.remove()
+    blossom_submission = response.data
+    response = cfg.blossom.unclaim(
+        submission_id=blossom_submission["id"], username=post.author.name
+    )
+    unclaim_messages = i18n["responses"]["unclaim"]
+    if response.status == BlossomStatus.ok:
+        message = unclaim_messages["success"]
+        flair_post(submission, flair.unclaimed)
+        removal, reported = _check_removal_required(submission, cfg)
+        if removal:
+            submission.mod.remove()
```

Comment:
this section seems like a good candidate for abstraction into a helper function.